backend/src/main/java/instantPolls/controller/RestControllerImpl.java

```diff
@@ -70,18 +67,29 @@
 	
 	@ResponseBody
 	@GetMapping(value = "/room/{id}")
-	public Room getRoom(@PathVariable(value = "id") String id) {
+	public ResponseEntity<Room> getRoom(@PathVariable(value = "id") String id) {
 		Room room = roomStorage.findRoomById(id);
-		return room;
+		if(room != null)
+			return ResponseEntity.ok(room);
+		else
+			return ResponseEntity.ok(null);
 	}
 	
 	@ResponseBody
-	@DeleteMapping(value = "/room/{id}/{token}")
-	public Map<String,String> closeRoom(@PathVariable(value = "id") String id,@PathVariable(value = "token") String token) {
-		if(roomStorage.closeRoom(id, token))
-			return Collections.singletonMap("result", "success");
-		else
-			return Collections.singletonMap("result", "failed");
+	@DeleteMapping(value = "/room/{id}")
+	public ResponseEntity<Map<String, String>> closeRoom(@PathVariable(value = "id") String id, @RequestHeader("Authorization") String token) {
+		int result = roomStorage.closeRoom(id, token);
+		if(result == 200) 
+			return ResponseEntity.ok(Collections.singletonMap("result", "success"));
+		else {
+			Map<String, String> resultString = Collections.singletonMap("result", "fail");	
+			if (result == 404)

```

Comment:
case would look better. Btw I don't like this 'pattern matching' here... I suppose that there may be some very similar copy paste around in the code...